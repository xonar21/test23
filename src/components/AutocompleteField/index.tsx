import React, { useState, useEffect } from "react";
import { FormControl, TextField, CircularProgress, Autocomplete, styled } from "@mui/material";
import { useDispatch } from "react-redux";
import { SelectListAction, SelectListSelector, SelectListsActions } from "~/store";

interface AutocompleteFieldProps {
  field: any;
  fieldValue: any;
  fieldKey: string;
  setValue: (newValue: any) => void;
  value: any;
  dinamicTitle: string;
  t: any;
  openModal: boolean;
  selectList: any;
  paramsRequest: any;
}

const AutocompleteField: React.FC<AutocompleteFieldProps> = ({
  field,
  fieldValue,
  fieldKey,
  setValue,
  value,
  dinamicTitle,
  t,
  openModal,
  selectList,
  paramsRequest,
}) => {
  const dispatch = useDispatch();
  const [page, setPage] = useState(1);
  const [loading, setLoading] = useState(false);
  const [totalPages, setTotalPages] = useState(0);
  const [searchValue, setSearchValue] = useState("");
  const actionKey: SelectListAction = field.action;
  const action = SelectListsActions[actionKey];
  const selectkey: SelectListSelector = field.keySelect;
  const [items, setItems] = useState(selectList[selectkey].items);
  const StyledListbox = styled("ul")({
    "&::-webkit-scrollbar": {
      width: "4px",
    },
    "&::-webkit-scrollbar-thumb": {
      backgroundColor: "rgba(0, 48, 92, 1)",
      borderRadius: "4px",
    },
    "&::-webkit-scrollbar-track": {
      backgroundColor: "rgba(230, 230, 230, 1)",
    },
  });

  useEffect(() => {
    const fetchData = async () => {
      if (field.action && !field.parent) {
        await dispatch(
          action(
            paramsRequest(page, 20, {
              fieldName: "value",
              values: [searchValue],
            }),
          ),
        );
      }
    };

    fetchData();
  }, [page, searchValue]);

  useEffect(() => {
    setTotalPages(selectList[selectkey].totalPages);

    if (selectList[selectkey].totalPages === 1) {
      setItems(selectList[selectkey].items);

      return;
    }

    setItems((prevItems: any) => [...prevItems, ...selectList[selectkey].items]);
  }, [selectList[selectkey].items]);

  useEffect(() => {
    const fetchData = async () => {
      if (field.parent && value[field.parent]) {
        const paramsRequest = (
          id: string,
          number?: number,
          size?: number,
          filters?: object,
          sortField?: string,
          sortOrder?: string,
        ) => {
          return {
            id,
            number,
            size,
            filters,
            sortField,
            sortOrder,
          };
        };

        await dispatch(
          action(
            paramsRequest(value[field.parent], page, 20, {
              fieldName: "value",
              values: [searchValue],
            }),
          ),
        );
      }
    };

    fetchData();
  }, [value[field.parent]]);

  useEffect(() => {
    if (field.parent) {
      setItems([]);
    }
  }, [openModal]);

  return (
    <FormControl fullWidth>
      <Autocomplete
        id="demo-autocomplete"
        options={items}
        value={fieldValue}
        getOptionLabel={option => {
          if (dinamicTitle === t("edit")) {
            let itemValue = "";
            items.forEach((e: any) => {
              if (e.key === option) {
                itemValue = e.value;
              }
            });
            return itemValue;
          }
          return option ? option.value : "";
        }}
        onChange={(e, newValue) => {
          setValue({ ...value, [fieldKey]: newValue?.key });
        }}
        onInputChange={(event, value, reason) => {
          if (reason === "reset") {
            setPage(1);
          }
          if (reason === "input") {
            setSearchValue(value);
            setPage(1);
            setItems([]);
          }
          if (reason === "clear") {
            setPage(1);
            setSearchValue("");
          }
          if (value === "") {
            setPage(1);
            setSearchValue("");
          }
        }}
        ListboxComponent={StyledListbox}
        ListboxProps={{
          onScroll: event => {
            const target = event.target as HTMLUListElement;
            const { scrollTop, scrollHeight, clientHeight } = target;
            if (scrollHeight - scrollTop === clientHeight) {
              setPage(prevPage => {
                if (prevPage !== totalPages) {
                  return prevPage + 1;
                }
                return prevPage;
              });
            }
          },
        }}
        renderInput={params => (
          <TextField
            {...params}
            label={t(`${field.fieldName}`)}
            InputLabelProps={{ ...params.InputLabelProps }}
            InputProps={{
              ...params.InputProps,
              endAdornment: (
                <>
                  {loading ? <CircularProgress size={20} /> : null}
                  {params.InputProps.endAdornment}
                </>
              ),
            }}
          />
        )}
        renderOption={(props, option) => <li {...props}>{option.value}</li>}
      />
    </FormControl>
  );
};

export default AutocompleteField;
