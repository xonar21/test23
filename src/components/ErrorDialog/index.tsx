import React from "react";
import Dialog from "@mui/material/Dialog";
import DialogTitle from "@mui/material/DialogTitle";
import DialogContent from "@mui/material/DialogContent";
import DialogContentText from "@mui/material/DialogContentText";
import DialogActions from "@mui/material/DialogActions";
import Button from "@mui/material/Button";

interface ErrorDialogProps {
  open: boolean;
  onClose: () => void;
  errorText: string;
}

const ErrorDialog: React.FC<ErrorDialogProps> = ({ open, onClose, errorText }) => {
  return (
    <Dialog open={open} onClose={onClose}>
      <DialogTitle>{"Ошибка"}</DialogTitle>
      <DialogContent>
        <DialogContentText>{errorText}</DialogContentText>
      </DialogContent>
      <DialogActions>
        <Button onClick={onClose} color="primary">
          Закрыть
        </Button>
      </DialogActions>
    </Dialog>
  );
};

export default ErrorDialog;
