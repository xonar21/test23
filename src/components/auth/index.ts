export * from "./auth-button";
export * from "./auth-form";
export * from "./auth-provider";
