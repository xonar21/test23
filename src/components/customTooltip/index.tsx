import React, { useMemo } from "react";
import { ColDef, ITooltipParams } from "ag-grid-community";
import { Box } from "@mui/material";

type CustomTooltipParams = ITooltipParams & {
  color: string;
  colDef: ColDef;
  shouldShowTooltip: boolean;
  rowData: any;
};
export default (props: CustomTooltipParams) => {
  const data = useMemo(() => props.api.getDisplayedRowAtIndex(props.rowIndex!)!.data, []);

  if (props && props.column) {
    if ("colId" in props.column) {
      const cellSelector = `.ag-row[row-id="${props.rowIndex}"] .ag-cell[col-id="${props.column.getColId()}"]`;
      const cellElement = document.querySelector(cellSelector);
      if (cellElement && cellElement) {
        if (cellElement.scrollWidth === cellElement.clientWidth) {
          return null;
        }
      }
    }
  }

  const subProperties = (data: any, props: any) => {
    const [firstField, secondField] = props.colDef.field.split(".");

    if (secondField) {
      return data[firstField][secondField];
    }

    return data[firstField];
  };
  return (
    <>
      {props.colDef.field ? (
        <Box
          sx={{
            backgroundColor: props.color || "white",
            border: "1px solid rgba(0, 48, 92, 1)",
            minWidth: "100px",
            borderRadius: "3px",
            padding: "5px",
            transition: "all 1s linear",
            maxWidth: "300px",
            whiteSpace: "pre-wrap !important",
            overflowWrap: "break-word !important",
            boxSizing: "border-box !important",
          }}
        >
          {subProperties(data, props)}
          {/* {data[props.colDef.field]} */}
        </Box>
      ) : (
        <></>
      )}
    </>
  );
};
