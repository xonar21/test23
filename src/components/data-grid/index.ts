export { default as DataGrid } from "./DataGrid";
export * from "./configs";
