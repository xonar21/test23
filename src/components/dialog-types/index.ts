export { default as DialogTypes } from "./dialog-types";
export * from "./dialog-types";
