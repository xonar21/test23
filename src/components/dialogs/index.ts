export * from "./prompt-dialog";
export * from "./post-form-dialog";
export * from "./dialog-provider";
