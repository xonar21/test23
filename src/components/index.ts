export * from "./auth";
export * from "./breadcrumbs";
export * from "./data-grid";
export * from "./demo";
export * from "./dialogs";
export * from "./dialog-types";
export * from "./formik";
export * from "./header";
export * from "./mui";
export * from "./navigation-drawer";
export * from "./page-loading-indicator";
export * from "./seo";
export * from "./snackbar-provider";
