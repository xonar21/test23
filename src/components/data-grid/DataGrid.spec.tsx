/* eslint-disable @typescript-eslint/no-explicit-any */
import { DataGrid } from "~/components";
import { fireEvent, render } from "~/shared";

const mockDispatch = jest.fn();
const mockSelectors = jest.fn();
const mockActions = {
  getList: jest.fn(),
  SELECTED_ROWS_CHANGED: jest.fn(),
  PAGE_CHANGED: jest.fn(),
};

jest.mock("react-redux", () => ({
  ...jest.requireActual("react-redux"),
  useDispatch: () => mockDispatch,
  useSelector: jest.fn(() => ({
    loading: false,
    limit: 5,
    page: 0,
    total: 10,
    data: [{ id: 1 }, { id: 2 }, { id: 3 }, { id: 4 }, { id: 5 }],
    selectedRows: [],
  })),
}));

describe("DataGrid component", () => {
  const columns = [
    {
      field: "id",
      headerName: "Id",
    },
  ];

  const grid = () => (
    <DataGrid
      paginationActions={mockActions as any}
      paginationSelectors={mockSelectors as any}
      columns={columns}
      checkboxSelection
    />
  );

  it("should match snapshot", () => {
    const { asFragment } = render(grid());
    expect(asFragment()).toMatchSnapshot();
  });

  it("should change page", () => {
    const { getByLabelText } = render(grid());
    const button = getByLabelText("Go to next page");
    fireEvent.click(button);
    expect(mockActions.PAGE_CHANGED).toBeCalled();
  });

  it("should select all rows", () => {
    const { getByLabelText } = render(grid());
    const checkbox = getByLabelText("Select all rows");
    fireEvent.click(checkbox);
    expect(mockActions.SELECTED_ROWS_CHANGED).toBeCalled();
  });
});
