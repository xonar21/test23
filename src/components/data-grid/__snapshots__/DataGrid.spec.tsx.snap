// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataGrid component should match snapshot 1`] = `
<DocumentFragment>
  <div
    aria-colcount="2"
    aria-multiselectable="false"
    aria-rowcount="11"
    class="MuiDataGrid-root MuiDataGrid-autoHeight MuiDataGrid-root--densityStandard css-1e2bxag-MuiDataGrid-root"
    role="grid"
  >
    <div>
      <div />
    </div>
    <div
      class="MuiDataGrid-main css-204u17-MuiDataGrid-main"
    >
      <div
        class="MuiDataGrid-columnHeaders css-z8fhq1-MuiDataGrid-columnHeaders"
        style="min-height: 56px; max-height: 56px; line-height: 56px;"
      >
        <div
          class="MuiDataGrid-columnHeadersInner MuiDataGrid-columnHeadersInner--scrollable css-gl260s-MuiDataGrid-columnHeadersInner"
          role="rowgroup"
          style="transform: translate3d(0px, 0px, 0px);"
        >
          <div
            aria-rowindex="1"
            class="css-yrdy0g-MuiDataGrid-columnHeaderRow"
            role="row"
          >
            <div
              aria-colindex="1"
              aria-sort="none"
              class="MuiDataGrid-columnHeader MuiDataGrid-columnHeader--alignCenter MuiDataGrid-columnHeaderCheckbox"
              data-field="__check__"
              role="columnheader"
              style="height: 56px; width: 50px; min-width: 50px; max-width: 50px;"
              tabindex="0"
            >
              <div
                class="MuiDataGrid-columnHeaderDraggableContainer"
                draggable="false"
              >
                <div
                  class="MuiDataGrid-columnHeaderTitleContainer"
                >
                  <div
                    class="MuiDataGrid-columnHeaderTitleContainerContent"
                  >
                    <span
                      class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-root MuiCheckbox-colorPrimary MuiDataGrid-checkboxInput css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
                    >
                      <input
                        aria-label="Select all rows"
                        class="PrivateSwitchBase-input css-1m9pwf3"
                        data-indeterminate="false"
                        tabindex="-1"
                        type="checkbox"
                      />
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="CheckBoxOutlineBlankIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </span>
                  </div>
                </div>
              </div>
              <div
                class="MuiDataGrid-columnSeparator MuiDataGrid-columnSeparator--sideRight"
                style="min-height: 56px; opacity: 1;"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiDataGrid-iconSeparator css-i4bv87-MuiSvgIcon-root"
                  data-testid="SeparatorIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M11 19V5h2v14z"
                  />
                </svg>
              </div>
            </div>
            <div
              aria-colindex="2"
              aria-label="Id"
              aria-sort="none"
              class="MuiDataGrid-columnHeader MuiDataGrid-columnHeader--sortable"
              data-field="id"
              role="columnheader"
              style="height: 56px; width: 100px; min-width: 100px; max-width: 100px;"
              tabindex="-1"
            >
              <div
                class="MuiDataGrid-columnHeaderDraggableContainer"
                draggable="false"
              >
                <div
                  class="MuiDataGrid-columnHeaderTitleContainer"
                >
                  <div
                    class="MuiDataGrid-columnHeaderTitleContainerContent"
                  >
                    <div
                      aria-label=""
                      class="MuiDataGrid-columnHeaderTitle css-1jbbcbn-MuiDataGrid-columnHeaderTitle"
                      data-mui-internal-clone-element="true"
                    >
                      Id
                    </div>
                  </div>
                  <div
                    class="MuiDataGrid-iconButtonContainer css-ltf0zy-MuiDataGrid-iconButtonContainer"
                  >
                    <button
                      aria-label="Sort"
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                      tabindex="-1"
                      title="Sort"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall MuiDataGrid-sortIcon css-ptiqhd-MuiSvgIcon-root"
                        data-testid="ArrowUpwardIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </div>
                <div
                  class="MuiDataGrid-menuIcon"
                >
                  <button
                    aria-controls="mui-6"
                    aria-haspopup="true"
                    aria-label="Menu"
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall MuiDataGrid-menuIconButton css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                    id="mui-7"
                    tabindex="-1"
                    title="Menu"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                      data-testid="TripleDotsVerticalIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                      />
                    </svg>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
              <div
                class="MuiDataGrid-columnSeparator MuiDataGrid-columnSeparator--sideRight"
                style="min-height: 56px; opacity: 1;"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiDataGrid-iconSeparator css-i4bv87-MuiSvgIcon-root"
                  data-testid="SeparatorIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M11 19V5h2v14z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        style="overflow: visible; width: 0px;"
      >
        <div
          class="MuiDataGrid-virtualScroller css-1w5m2wr-MuiDataGrid-virtualScroller"
          style="width: 0px; height: auto; margin-top: 56px; overflow-x: hidden; overflow-y: hidden;"
        >
          <div
            class="MuiDataGrid-virtualScrollerContent css-1kwdphh-MuiDataGrid-virtualScrollerContent"
            style="width: auto; height: 260px; min-height: auto;"
          >
            <div
              class="MuiDataGrid-virtualScrollerRenderZone css-s1v7zr-MuiDataGrid-virtualScrollerRenderZone"
              style="transform: translate3d(0px, 0px, 0px);"
            >
              <div
                aria-rowindex="2"
                aria-selected="false"
                class="MuiDataGrid-row"
                data-id="1"
                data-rowindex="0"
                role="row"
                style="max-height: 52px; min-height: 52px;"
              >
                <div
                  aria-colindex="1"
                  aria-colspan="1"
                  class="MuiDataGrid-cellCheckbox MuiDataGrid-cell--withRenderer MuiDataGrid-cell MuiDataGrid-cell--textCenter"
                  data-colindex="0"
                  data-field="__check__"
                  role="cell"
                  style="min-width: 50px; max-width: 50px; min-height: 52px; max-height: 52px;"
                  tabindex="-1"
                >
                  <span
                    class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-root MuiCheckbox-colorPrimary MuiDataGrid-checkboxInput css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
                  >
                    <input
                      aria-label="Select row"
                      class="PrivateSwitchBase-input css-1m9pwf3"
                      data-indeterminate="false"
                      tabindex="-1"
                      type="checkbox"
                    />
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="CheckBoxOutlineBlankIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                      />
                    </svg>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </span>
                </div>
                <div
                  aria-colindex="2"
                  aria-colspan="1"
                  class="MuiDataGrid-cell MuiDataGrid-cell--textLeft"
                  data-colindex="1"
                  data-field="id"
                  role="cell"
                  style="min-width: 100px; max-width: 100px; min-height: 52px; max-height: 52px;"
                  tabindex="-1"
                >
                  <div
                    class="MuiDataGrid-cellContent"
                    title="1"
                  >
                    1
                  </div>
                </div>
              </div>
              <div
                aria-rowindex="3"
                aria-selected="false"
                class="MuiDataGrid-row"
                data-id="2"
                data-rowindex="1"
                role="row"
                style="max-height: 52px; min-height: 52px;"
              >
                <div
                  aria-colindex="1"
                  aria-colspan="1"
                  class="MuiDataGrid-cellCheckbox MuiDataGrid-cell--withRenderer MuiDataGrid-cell MuiDataGrid-cell--textCenter"
                  data-colindex="0"
                  data-field="__check__"
                  role="cell"
                  style="min-width: 50px; max-width: 50px; min-height: 52px; max-height: 52px;"
                  tabindex="-1"
                >
                  <span
                    class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-root MuiCheckbox-colorPrimary MuiDataGrid-checkboxInput css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
                  >
                    <input
                      aria-label="Select row"
                      class="PrivateSwitchBase-input css-1m9pwf3"
                      data-indeterminate="false"
                      tabindex="-1"
                      type="checkbox"
                    />
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="CheckBoxOutlineBlankIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                      />
                    </svg>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </span>
                </div>
                <div
                  aria-colindex="2"
                  aria-colspan="1"
                  class="MuiDataGrid-cell MuiDataGrid-cell--textLeft"
                  data-colindex="1"
                  data-field="id"
                  role="cell"
                  style="min-width: 100px; max-width: 100px; min-height: 52px; max-height: 52px;"
                  tabindex="-1"
                >
                  <div
                    class="MuiDataGrid-cellContent"
                    title="2"
                  >
                    2
                  </div>
                </div>
              </div>
              <div
                aria-rowindex="4"
                aria-selected="false"
                class="MuiDataGrid-row"
                data-id="3"
                data-rowindex="2"
                role="row"
                style="max-height: 52px; min-height: 52px;"
              >
                <div
                  aria-colindex="1"
                  aria-colspan="1"
                  class="MuiDataGrid-cellCheckbox MuiDataGrid-cell--withRenderer MuiDataGrid-cell MuiDataGrid-cell--textCenter"
                  data-colindex="0"
                  data-field="__check__"
                  role="cell"
                  style="min-width: 50px; max-width: 50px; min-height: 52px; max-height: 52px;"
                  tabindex="-1"
                >
                  <span
                    class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-root MuiCheckbox-colorPrimary MuiDataGrid-checkboxInput css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
                  >
                    <input
                      aria-label="Select row"
                      class="PrivateSwitchBase-input css-1m9pwf3"
                      data-indeterminate="false"
                      tabindex="-1"
                      type="checkbox"
                    />
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="CheckBoxOutlineBlankIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                      />
                    </svg>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </span>
                </div>
                <div
                  aria-colindex="2"
                  aria-colspan="1"
                  class="MuiDataGrid-cell MuiDataGrid-cell--textLeft"
                  data-colindex="1"
                  data-field="id"
                  role="cell"
                  style="min-width: 100px; max-width: 100px; min-height: 52px; max-height: 52px;"
                  tabindex="-1"
                >
                  <div
                    class="MuiDataGrid-cellContent"
                    title="3"
                  >
                    3
                  </div>
                </div>
              </div>
              <div
                aria-rowindex="5"
                aria-selected="false"
                class="MuiDataGrid-row"
                data-id="4"
                data-rowindex="3"
                role="row"
                style="max-height: 52px; min-height: 52px;"
              >
                <div
                  aria-colindex="1"
                  aria-colspan="1"
                  class="MuiDataGrid-cellCheckbox MuiDataGrid-cell--withRenderer MuiDataGrid-cell MuiDataGrid-cell--textCenter"
                  data-colindex="0"
                  data-field="__check__"
                  role="cell"
                  style="min-width: 50px; max-width: 50px; min-height: 52px; max-height: 52px;"
                  tabindex="-1"
                >
                  <span
                    class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-root MuiCheckbox-colorPrimary MuiDataGrid-checkboxInput css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
                  >
                    <input
                      aria-label="Select row"
                      class="PrivateSwitchBase-input css-1m9pwf3"
                      data-indeterminate="false"
                      tabindex="-1"
                      type="checkbox"
                    />
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="CheckBoxOutlineBlankIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                      />
                    </svg>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </span>
                </div>
                <div
                  aria-colindex="2"
                  aria-colspan="1"
                  class="MuiDataGrid-cell MuiDataGrid-cell--textLeft"
                  data-colindex="1"
                  data-field="id"
                  role="cell"
                  style="min-width: 100px; max-width: 100px; min-height: 52px; max-height: 52px;"
                  tabindex="-1"
                >
                  <div
                    class="MuiDataGrid-cellContent"
                    title="4"
                  >
                    4
                  </div>
                </div>
              </div>
              <div
                aria-rowindex="6"
                aria-selected="false"
                class="MuiDataGrid-row MuiDataGrid-row--lastVisible"
                data-id="5"
                data-rowindex="4"
                role="row"
                style="max-height: 52px; min-height: 52px;"
              >
                <div
                  aria-colindex="1"
                  aria-colspan="1"
                  class="MuiDataGrid-cellCheckbox MuiDataGrid-cell--withRenderer MuiDataGrid-cell MuiDataGrid-cell--textCenter"
                  data-colindex="0"
                  data-field="__check__"
                  role="cell"
                  style="min-width: 50px; max-width: 50px; min-height: 52px; max-height: 52px;"
                  tabindex="-1"
                >
                  <span
                    class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-root MuiCheckbox-colorPrimary MuiDataGrid-checkboxInput css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
                  >
                    <input
                      aria-label="Select row"
                      class="PrivateSwitchBase-input css-1m9pwf3"
                      data-indeterminate="false"
                      tabindex="-1"
                      type="checkbox"
                    />
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="CheckBoxOutlineBlankIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                      />
                    </svg>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </span>
                </div>
                <div
                  aria-colindex="2"
                  aria-colspan="1"
                  class="MuiDataGrid-cell MuiDataGrid-cell--textLeft"
                  data-colindex="1"
                  data-field="id"
                  role="cell"
                  style="min-width: 100px; max-width: 100px; min-height: 52px; max-height: 52px;"
                  tabindex="-1"
                >
                  <div
                    class="MuiDataGrid-cellContent"
                    title="5"
                  >
                    5
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="Mui-resizeTriggers"
      >
        <div
          class="expand-trigger"
        >
          <div
            style="width: 1px; height: 1px;"
          />
        </div>
        <div
          class="contract-trigger"
        />
      </div>
    </div>
    <div>
      <div
        class="MuiDataGrid-footerContainer css-17jjc08-MuiDataGrid-footerContainer"
      >
        <div />
        <div
          class="MuiTablePagination-root css-rtrcn9-MuiTablePagination-root"
        >
          <div
            class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular MuiTablePagination-toolbar css-78c6dr-MuiToolbar-root-MuiTablePagination-toolbar"
          >
            <div
              class="MuiTablePagination-spacer css-1psng7p-MuiTablePagination-spacer"
            />
            <p
              class="MuiTablePagination-selectLabel css-pdct74-MuiTablePagination-selectLabel"
              id="mui-2"
            >
              Rows per page:
            </p>
            <div
              class="MuiInputBase-root MuiInputBase-colorPrimary css-16c50h-MuiInputBase-root-MuiTablePagination-select"
            >
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby="mui-2 mui-1"
                class="MuiSelect-select MuiTablePagination-select MuiSelect-standard MuiInputBase-input css-194a1fa-MuiSelect-select-MuiInputBase-input"
                id="mui-1"
                role="button"
                tabindex="0"
              >
                5
              </div>
              <input
                aria-hidden="true"
                class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                tabindex="-1"
                value="5"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiTablePagination-selectIcon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
            <p
              class="MuiTablePagination-displayedRows css-levciy-MuiTablePagination-displayedRows"
            >
              1–5 of 10
            </p>
            <div
              class="MuiTablePagination-actions"
            >
              <button
                aria-label="Go to previous page"
                class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
                disabled=""
                tabindex="-1"
                title="Go to previous page"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="KeyboardArrowLeftIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                  />
                </svg>
              </button>
              <button
                aria-label="Go to next page"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                title="Go to next page"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="KeyboardArrowRightIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
